lib/common/client/middleware/logging.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Common
+  module Client
+    module Middleware
+      class Logging < Faraday::Middleware
+        DEFAULT_TYPE_KEY = 'undefined'
+
+        def initialize(app, type_key)
+          super(app)
+          @type_key = type_key || DEFAULT_TYPE_KEY
+        end
+
+        def call(env)
+          request_body = Base64.encode64(env.body)
+
+          @app.call(env).on_complete do |response_env|
+            PersonalInformationLog.create({
+              error_class: @type_key, # TODO: error_class is probably worth renaming
```

Comment:
`error_class` was the original name for the lookup key. Super minor, so I think a TODO in the codebase is warranted but not necessarily a new vets.gov-team issue (since we have about +2K of those already that are all likely a higher priority than this).